src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -146,31 +147,49 @@ def _create_rpc_error(initial_metadata: Optional[MetadataType],
 
 
 class Call(_base_call.Call):
+    """Base implementation of client RPC Call object.
+
+    Implements logic around final status, metadata and cancellation.
+    """
     _loop: asyncio.AbstractEventLoop
     _code: grpc.StatusCode
     _status: Awaitable[cygrpc.AioRpcStatus]
     _initial_metadata: Awaitable[MetadataType]
     _locally_cancelled: bool
+    _cython_call: cygrpc._AioCall
 
-    def __init__(self) -> None:
+    def __init__(self, deadline: Optional[float], channel: cygrpc.AioChannel,
+                 method: bytes) -> None:
         self._loop = asyncio.get_event_loop()
         self._code = None
         self._status = self._loop.create_future()
         self._initial_metadata = self._loop.create_future()
         self._locally_cancelled = False
+        self._cython_call = channel.call(method, deadline)
 
-    def cancel(self) -> bool:
-        """Placeholder cancellation method.
-
-        The implementation of this method needs to pass the cancellation reason
-        into self._cancellation, using `set_result` instead of
-        `set_exception`.
-        """
-        raise NotImplementedError()
+    def __del__(self) -> None:
+        if not self._status.done():
+            self._cancel(
+                cygrpc.AioRpcStatus(cygrpc.StatusCode.cancelled,
+                                    _GC_CANCELLATION_DETAILS, None, None))
 
     def cancelled(self) -> bool:
         return self._code == grpc.StatusCode.CANCELLED
 
+    def _cancel(self, status: cygrpc.AioRpcStatus) -> bool:
+        """Forwards the application cancellation reasoning."""
+        if not self._status.done():
+            self._set_status(status)
+            self._cython_call.cancel(status)
+            return True
+        else:
+            return False
+
+    def cancel(self) -> bool:
```

Comment:
You're right. I did miss that. The docstring says "A bool indicates if the cancellation is performed or not." That's not quite what I'm seeing in the code here though. As currently written, I would assume that `False` means cancellation failed. What I'm seeing here is a `True` value means that this particular invocation resulted in active cancellation of the RPC, whereas a value of `False` implies that no work was performed to cancel this RPC as it was already cancelled.

I'm now wondering what the use case for the return value of this function is. Perhaps we should just document that it's idempotent and return no value at all. In case there is an actual problem cancelling the RPC, we can raise an exception.

Reply:
Discussed offline. Keeping the current interface because it matches up with the `asyncio.Future.cancel()` signature.